recipes/gcTester_validate_crc_mismatch_with_s3.yml

```diff
@@ -33,62 +33,83 @@
 
     - name: "{{ recipe_name }}: upload corrupted file"
       vars:
-        Chunk: "{{ chunk['msg'][0] }}"
+        formal_params: {
+                "chunk" : "{{ chunk['msg'][0] }}",
+                "path": "{{ paths['msg'][0] }}",
+                }
       debug:
-         msg: "{{ lookup('obj_storage', 'operation', 'upload', Chunk, paths['msg'][0], wantlist=True) }}"
+         msg: "{{ lookup('obj_storage', 'operation', 'upload', formal_params, wantlist=True) }}"
 
     - name: "{{ recipe_name }}: Perform GC."
       vars:
-        debugMode: false
-        Chunk: "{{ chunk['msg'][0] }}"
-        crcCheck: true
+        input_params:
+          debug_mode: "false"
+          chunk: "{{ chunk['msg'][0] }}"
+          crc_check: "true"
       debug:
-         msg:
-          - "{{ lookup('s3_handler', 'start_gc', debugMode, Chunk, crcCheck, wantlist=True) }}"
+         msg: "{{ lookup('garbage_collection', 'single_exec', 'start', input_params, wantlist=True) }}"
 
     - name: "{{ recipe_name }}: get gc and nisd markers"
       vars:
-        Chunk: "{{ chunk['msg'][0] }}"
+        formal_params: {
+                "chunk" : "{{ chunk['msg'][0] }}",
+                }
       debug:
-         msg: "{{ lookup('obj_storage', 'get_markers', Chunk, wantlist=True) }}"
+         msg: "{{ lookup('obj_storage', 'get_markers', formal_params, wantlist=True) }}"
       register: marker_check
 
     - name: "{{ recipe_name }}: compare gc and nisd markers"
       set_fact:
         gc_marker: "{{ marker_check.msg[0] }}"
         nisd_marker: "{{ marker_check.msg[1] }}"
-      when: marker_check.msg is defined and marker_check.msg[0] | selectattr('!=', 'none') 
+      when: marker_check.msg is defined and marker_check.msg[0] != 'none'
 
     - debug:
          msg: "{{gc_marker}} and {{nisd_marker}}"
       when: gc_marker is defined and nisd_marker is defined
       failed_when: gc_marker != nisd_marker
     
```

Comment:
do we need a step here to validate if the corrupted file is skipped by gc or not?

Reply:
@RAVAN0407